### ISSUE TITLE
Vaadin 8

### ISSUE THREAD

**Issue by nikkijuk** [melistik/vaadin-grid-util#37]
Is is planned that this add on would be ported to vaadin 8?

- Jukka

**Comment by kabrender** [comment]
Hello,
is a port to vaadin 8 in planing?

**Comment by melistik** [comment]
Yes is in planning - i've already started but it's a huge rebuild because the main features of Container-Filtering is getting removed :)

**Comment by ossd** [comment]
Is it possible to create a new version with vaadin 8 compatibility package for migration and after that a new version for vaadin 8 grid?

**Comment by melistik** [comment]
hi, currently the migration is not finished - I have posted some issues to vaadin. their api changes don't allow a normal use as designed in v7. Thats why it will take propably some weeks to get everything running again...

**Comment by nikkijuk** [comment]
Is vaadin currently containing two grids? v7 and v8 versions? I guess so.. That must create some mess on communication. I have used only V8 version and they've done there there a lot of changes - I wouldn't be suprised if they still have something to fix / redesign even if I think they've done good job and implementation has been pretty stable.

**Comment by melistik** [comment]
I've started with a separate branch [vaadin-8-migration](https://github.com/melistik/vaadin-grid-util/tree/vaadin-8-migration) - just to use the v7 version. But I still got some issues. Maybe someone can support me.

I get some JavaScript exceptions (so the Renders don't work correctly) and other columns expect birthDay didn't get displayed (maybe due to the previous error)...

**Comment by PhineasClick** [comment]
Hi melistik, can we may support you by somehow upvoting for the issues you openend with vaadin?

**Comment by pethers** [comment]
Happy to help out as well, really miss this plugin when I migrated partly to vaadin8 using com.vaadin.v7 packages. Hard to find a politician in https://www.hack23.com/cia/#!politicianranking/DATAGRID without any filters. Would be awesome with vaadin8 support.

**Comment by pethers** [comment]
Forked and created a branch https://github.com/pethers/vaadin-grid-util/tree/vaadin8-v7-support , using vaadin-compatibility-server with vaadin7 packages moved to v7.  Commented out some code in org/vaadin/gridutil/client/renderer/buttonvalue/AbstractButtonValueConnector.java

//				getRpcProxy(RendererClickRpc.class).click(getRowKey(event.getCell()
//						.getRow()), getColumnId(event.getCell()
//						.getColumn()), details);

Doing a soft migration to vaadin8 using v7 packages, so above branch will work for a while.

**Comment by melistik** [comment]
Hey guys,

currently im quite busy and don't use vaadin-8 in any business case that's why I currently cannot help you out. Sorry :(

@pethers I've testes your branch I seams to work but I still get JavaScript errors for each column:
```stacktrace
com.vaadin.client.widgets.Grid
SEVERE: Error rendering cell in column 3
com.google.gwt.core.client.JavaScriptException: (TypeError) : Cannot read property 'v9b' of null
```

Could your maybe fix it and afterwards I would build an alpha release out of it.

@PhineasClick thanks for your offer but for me vaadin has changed too many thing. This projects has next to migration-version that's not working also a clean upgrade version (only vaadin-8) but in both cases I got stucked. In both cases I got some errors I don't find out what to change... Maybe you can run one of my brunch and ask vaadin to help out.

**Comment by nikkijuk** [comment]
If one browses release notes of 8.1 alphas & 8.0.X it starts to be evident that Grid is having severe issues with stability, compatibility and new features are coming in which eventually could extend Grid utils functionality.

https://github.com/vaadin/framework/releases

My 2 cents: it could be wise to fork Grid-Util as version 2.0 for vaadin 8.1 - no compatibility libraries - and give it bit time to stabilize and simultaneously try to get existing Grid-Util to function with vaadin 8.X + compatibility libraries.

**Comment by pethers** [comment]
@melistik will try to resolve the JavaScriptException, problem is that I don't get any errors myself. 

Don't see any exception when running selenium test (https://www.hack23.com/jenkins/job/Citizen-Intelligence-Agency-System-Test/2862/consoleFull) or starting vaadin-grid-util-demo.

**Comment by PhineasClick** [comment]
Are there still any efforts to port this addon to 8.0?

**Comment by melistik** [comment]
new release published 2.1.0
great thanks to @hitchhiker-jiri  he has done a great job
